t/integration/test_canvas.py

```diff
@@ -2020,7 +2167,45 @@ def test_errback_called_by_chord_from_group(self, manager, subtests):
             await_redis_echo({errback_msg, }, redis_key=redis_key)
         redis_connection.delete(redis_key)
 
-    def test_errback_called_by_chord_from_group_fail_multiple(
+    @pytest.mark.parametrize(
+        "errback_task", [errback_old_style, errback_new_style, ],
+    )
+    def test_mutable_errback_called_by_chord_from_group(
+        self, errback_task, manager, subtests
+    ):
+        if not manager.app.conf.result_backend.startswith("redis"):
+            raise pytest.skip("Requires redis result backend.")
+        redis_connection = get_redis_connection()
+
+        errback = errback_task.s()
+        fail_sig = fail.s()
+        fail_sig_id = fail_sig.freeze().id
+        child_sig = group(identity.si(42), fail_sig)
+
+        chord_sig = chord((child_sig, ), identity.s())
+        chord_sig.link_error(errback)
+        expected_redis_key = chord_sig.body.freeze().id
+        redis_connection.delete(expected_redis_key)
+        with subtests.test(msg="Error propagates from header group"):
+            res = chord_sig.delay()
+            with pytest.raises(ExpectedException):
+                res.get(timeout=TIMEOUT)
```

Comment:
This and the `res.get()` in the subtest below seem to sometimes time out and I can't work out why. It seems to only really occur when I run the full suite (redis backend) or run `tox -e ... -- -k test_canvas` to select the canvas integration tests.